packages/core/src/interfaces/components.ts

```diff
@@ -44,6 +45,18 @@ export interface LegendOptions {
 	 * customized legend items
 	 */
 	additionalItems?: LegendItem[];
+	/**
+	 * customize color legend
+	 * enabled by default on select charts
+	 */
+	colorLegend?: {
+		/**
+		 * Text to display beside or on top of the legend
+		 * Position is determined by text length
+		 */
+		title?: string;
+		type: ColorLegendType;
+	};
```

Comment:
would this add `colorLegend` to the types for the rest of the charts? is that what we want?

Reply:
The option currently is **ONLY** used by Heatmap - if we are going to have future charts that use color legend, it makes sense to put it under the legend option, since it is a legend. 

Setting the ``type`` or ``title`` will not do enable or do anything in other chart types.

Reply:
Let's (for now) extend this to a `HeatmapLegendOptions` and only include there, as we haven't figured out how this new legend type will work with other chart types